recipe-client-addon/lib/NormandyDriver.jsm

```diff
@@ -156,6 +156,47 @@ this.NormandyDriver = function(sandboxManager) {
       sandboxManager.removeHold(`setTimeout-${token}`);
     },
 
+    /*
+     * Return a promise that resolves to an Addon ID if installation is successful.
+     */
+    async installAddon(installUrl) {
+      const installObj = await AddonManager.getInstallForURL(installUrl, null, "application/x-xpinstall");
+      const result = new Promise((resolve, reject) => installObj.addListener({
+        onInstallEnded(addonInstall, addon) {
+          resolve(addon.id);
+        },
+        onInstallFailed(addonInstall) {
+          reject(`AddonInstall error code: [${addonInstall.error}]`);
+        },
+      }));
+      installObj.install();
+      return result;
+    },
+
+    /*
+     * Return a promise that resolves to a success messsage if
+     * addon uninstall is successful.
+     */
+    async uninstallAddon(addonId) {
+      return (id => {
```

Comment:
Concurring, as Mossop said earlier (in a now-collapsed-on-github bit), getAddonByID() returns a promise as well, so it should be possible to just do:

```js
let addon = await ...getAddonByID(addonId);
// do uninstall
```

to simplify this significantly.